src/ui/components/local-time-block.svelte

```diff
@@ -47,6 +49,21 @@
   });
 
   const { isActive: isResizeFromTopActive } = resizeFromTop;
+
+  $: override = useColorOverride(task);
+  // todo: hide in hook
+  $: backgroundColor =
+    $override || "var(--time-block-bg-color, var(--background-primary))";
+
+  $: ({
+    faint,
+    muted,
+    normal,
+  } = $override 
+    ? getTextColorWithEnoughContrast(backgroundColor)
+    : { faint: "", muted: "", normal: "" }
+  );
```

Comment:
This solution is simple, and it works, but let's keep all the color related code in one place. This should go into `src/ui/hooks/use-color.ts`. There, the same calculation is made for a different feature: colored timelines. We should check if a task matches an override there, and return the override colors. Overrides should take precedence over colored timeline colors.